packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/App.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.mongodb
+
+import io.realm.mongodb.internal.AppConfigurationImpl
+import io.realm.mongodb.internal.AppImpl
+import kotlinx.coroutines.CoroutineDispatcher
+
+/**
+ * TODO
+ */
+interface App {
+
+    val configuration: AppConfiguration
+
+    suspend fun login(credentials: Credentials): Result<User>
```

Comment:
We need to discuss the `Results` API used here, but no need to block the PR for it, but we should add a FIXME about it.

Personally, I think we might need to duplicate the stdlib Result class to something like `AppResult`, just so we can control the error case better. But it is annoying that it ends up being a copy just for the sake of specializing the Error case.

Reply:
Added a FIXME to revisit this with link to this comment.